plugins/task-plugin/src/machine/machines-picker.ts

```diff
@@ -29,24 +31,78 @@ export class MachinesPicker {
      */
     async pick(containerNames?: string[]): Promise<string> {
         if (!containerNames) {
-            containerNames = await this.cheWorkspaceClient.getContainersNames();
+            return this.pickContainerFromClient();
         }
 
-        if (containerNames.length === 1) {
-            return Promise.resolve(containerNames[0]);
+        return this.pickContainerFromStringList(containerNames);
+    }
+
+    private pickContainerFromStringList(containers: string[]): Promise<string> {
+        if (containers.length === 1) {
+            return Promise.resolve(containers[0]);
         }
-        return this.showMachineQuickPick(containerNames);
+        return this.showContainersStringQuickPick(containers);
```

Comment:
If I'm not mistaken you have two methods with exactly alike stuffing: `showContainersStringQuickPick()` and `showContainersQuickPick()`.
Can we avoid duplicated code fragment?

Reply:
Fixed